lapce-data/src/editor.rs

```diff
@@ -3562,10 +3562,23 @@ impl KeyPressFocus for LapceEditorBufferData {
             }
             LapceCommand::DeleteToBeginningOfLine => {
                 let selection = match self.editor.cursor.mode {
-                    CursorMode::Normal(_) | CursorMode::Visual { .. } => self
-                        .editor
-                        .cursor
-                        .edit_selection(&self.buffer, self.config.editor.tab_width),
+                    CursorMode::Normal(_) | CursorMode::Visual { .. } => {
+                        let selection = self.editor.cursor.edit_selection(
+                            &self.buffer,
+                            self.config.editor.tab_width,
+                        );
+
+                        self.buffer.update_selection(
+                            &selection,
+                            1,
+                            &Movement::StartOfLine,
+                            Mode::Insert,
+                            true,
+                            self.editor.code_lens,
+                            self.editor.compare.clone(),
+                            &self.config,
+                        )
+                    }

```

Comment:
Don't know if this changes any critical behaviour. Might be worth a check by someone who knows what the original code was for